csharp/ql/src/Diagnostics/DiagnosticExtractionErrors.ql

```diff
@@ -0,0 +1,55 @@
+/**
+ * @name Extraction errors
+ * @description List all errors reported by the extractor or the compiler. Extractor errors are
+ *              limited to those files where there are no compilation errors.
+ * @kind diagnostic
+ * @id cs/diagnostics/extraction-errors
+ */
+
+import csharp
+import semmle.code.csharp.commons.Diagnostics
+
+private newtype TDiagnosticError =
+  TCompilerError(CompilerError c) or
+  TExtractorError(ExtractorError e)
+
+abstract private class DiagnosticError extends TDiagnosticError {
+  string getMessage() { none() }
+
+  string toString() { none() }
+
+  string getLocation(Location l) {
```

Comment:
add `abstract Location getLocation();` and use that here instead of the parameter `l` (the latter has a hidden Cartesian product with `this`).